narwhal/types/src/primary.rs

```diff
@@ -717,7 +725,12 @@ impl HeaderV2 {
         *self.digest.get_or_init(|| Hash::digest(self))
     }
 
-    pub fn validate(&self, committee: &Committee, worker_cache: &WorkerCache) -> DagResult<()> {
+    pub fn validate(
+        &self,
+        committee: &Committee,
+        worker_cache: &WorkerCache,
+        randomness_vss_key: Option<&fastcrypto_tbls::types::PublicVssKey>,
```

Comment:
When after 2f+1 validators vote on a header and produce a signature each, the assumption is that the certificate created by aggregating the signatures can be verified by all honest validators. Any additional logic that may result in verification failure can cause protocol problems. So we should move randomness verifications to the request_vote handler. 

Also Certificate verification includes verifying the header digest is signed by >= 2f+1 validators. So if verifying randomness signature is done at voting time, that should be sufficient too.